Portal/vip-application/src/main/java/fr/insalyon/creatis/vip/application/server/business/WorkflowBusiness.java

```diff
@@ -113,6 +115,31 @@ public WorkflowBusiness() {
         }
     }
 
+    private String selectEngineEndpoint(String applicationClass) throws BusinessException {
```

Comment:
A problem I see with this selection method is that a failing engine may become a black hole receiving all the workflows and crashing them as soon as they are launched. The number of running workflows on such an engine would always be lower than on the other ones. For this reason, random or round-robin selection seems wiser. We should also think of a way to enable/disable engines in the interface (manually), and to automatically disable engines when they obviously fail, e.g. connection fails (workflow cannot be even submitted, we don't get an id back). Exponential backoff seems to be a good solution to enable/disable engines automatically. That could be done by adding a few parameters to the engines table (number of failures since last success, timestamp of last success, timestamp of last attempt).